src/Analysis/Ast/Impl/Caching/StubCache.cs

```diff
@@ -27,13 +27,15 @@ internal sealed class StubCache : IStubCache {
 
         private readonly IFileSystem _fs;
         private readonly ILogger _log;
+        private readonly ICacheFolderService _cfs;
 
-        public StubCache(IServiceContainer services, string cacheRootFolder = null) {
+        public StubCache(IServiceContainer services) {
             _fs = services.GetService<IFileSystem>();
             _log = services.GetService<ILogger>();
+            _cfs = services.GetService<ICacheFolderService>();
 
-            cacheRootFolder = cacheRootFolder ?? CacheFolders.GetCacheFolder(services);
-            StubCacheFolder = Path.Combine(cacheRootFolder, $"stubs.v{_stubCacheFormatVersion}");
+            _cfs = services.GetService<ICacheFolderService>();
```

Comment:
`ICacheFolderService` is accessed twice.

Reply:
Fixed